hack/heap/hh_shared.c

```diff
@@ -148,7 +148,7 @@
  * with the MAP_ANONYMOUS flag. The memfd_create() system call first
  * appeared in Linux 3.17.
  ****************************************************************************/
-#if !defined __APPLE__ && !defined _WIN32
+#ifdef __linux__
```

Comment:
If you look below where `MEMFD_CREATE` is checked, you'll see the non-apple case will fall through to a pretty terrible create-a-file-on-tmpfs code. FreeBSD seems to have [`shm_open` and `shm_unlink`](https://www.freebsd.org/cgi/man.cgi?query=shm_open&sektion=2&manpath=FreeBSD+8.4-RELEASE) so I wonder if they can share the apple implementation, which also uses those APIs.

But this is fine for now. Falling through to a terrible implementation is better than not working at all!